apitally/litestar.py

```diff
@@ -0,0 +1,176 @@
+
+        app_info = {
+            "openapi": _get_openapi(app),
+            "paths": [route for route in _get_routes(app) if not self.filter_path(route["path"])],
+            "versions": _get_versions(self.app_version),
+            "client": "python:litestar",
+        }
+        self.client.set_app_info(app_info)
+        self.client.start_sync_loop()
+
+    def middleware_factory(self, app: ASGIApp) -> ASGIApp:
+        async def middleware(scope: Scope, receive: Receive, send: Send) -> None:
+            if scope["type"] == "http" and scope["method"] != "OPTIONS":
+                request = Request(scope)
+                response_status = 0
+                response_time = 0.0
+                response_headers = Headers()
+                response_body = b""
+                start_time = time.perf_counter()
+
+                async def send_wrapper(message: Message) -> None:
+                    nonlocal response_time, response_status, response_headers, response_body
+                    if message["type"] == "http.response.start":
+                        response_time = time.perf_counter() - start_time
+                        response_status = message["status"]
+                        response_headers = Headers(message["headers"])
+                    elif message["type"] == "http.response.body" and response_status == 400:
+                        response_body += message["body"]
+                    await send(message)
+
+                await app(scope, receive, send_wrapper)
+                self.add_request(
+                    request=request,
+                    response_status=response_status,
+                    response_time=response_time,
+                    response_headers=response_headers,
+                    response_body=response_body,
+                )
+            else:
+                await app(scope, receive, send)  # pragma: no cover
+
+        return middleware
+
+    def add_request(
+        self,
+        request: Request,
+        response_status: int,
+        response_time: float,
+        response_headers: Headers,
+        response_body: bytes,
+    ) -> None:
+        if not request.route_handler.paths:
+            return  # pragma: no cover
+        path = list(request.route_handler.paths)[0]
+        if self.filter_path(path):
+            return
+        consumer = self.get_consumer(request)
+        self.client.request_counter.add_request(
+            consumer=consumer,
+            method=request.method,
+            path=path,
+            status_code=response_status,
+            response_time=response_time,
+            request_size=request.headers.get("Content-Length"),
+            response_size=response_headers.get("Content-Length"),
+        )
+        if response_status == 400 and response_body and len(response_body) < 4096:
+            try:
+                parsed_body = json.loads(response_body)
+            except json.JSONDecodeError:  # pragma: no cover
+                pass
```

Comment:
```suggestion
with contextlib.suppress
      parsed_body = json.loads(response_body)
```
saves a line (needs import contextlib though)

Reply:
I like that